Bottom: 5d1fb968a5ab0cdae1eebe2a78f192ff68b593d5
Top:    25dd9bb29e1780999684aa2ab5f943f27132604d
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2014-12-27 19:33:36 +0000

drc: HACK HACK HACK


---

diff --git a/src/find.c b/src/find.c
index c402568..14590b5 100644
--- a/src/find.c
+++ b/src/find.c
@@ -2460,7 +2460,12 @@ IsPolygonInPolygon (PolygonType *P1, PolygonType *P2)
 
               line.Point1.X = v->point[0];
               line.Point1.Y = v->point[1];
-              line.Thickness = 2 * Bloat;
+              line.Thickness = Bloat;
+              /* Another Bloat is added by IsLineInPolygon, making the correct
+               * 2x Bloat. Ideally we would change it there, but doing so
+               * breaks some other DRC checks which rely on the broken behaviour
+               * in IsLineInPolygon.
+               */
               line.Clearance = 0;
               line.Flags = NoFlags ();
               for (v = v->next; v != &c->head; v = v->next)
